client/src/app/pages/auth-page/auth-page.component.ts

```diff
@@ -0,0 +1,146 @@
+import { Apollo } from 'apollo-angular';
+import { TranslateService } from '@ngx-translate/core';
+import gql from 'graphql-tag';
+import { Component, OnInit, OnDestroy } from '@angular/core';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+import { ActivatedRoute } from '@angular/router';
+import { Router } from '@angular/router';
+
+import { AuthService } from '../../services/auth.service';
+import { ROLES } from 'src/enums/roles';
+import { EmailValidator } from '../../validators/email-correct.validator';
+import { Subscription } from 'rxjs';
+import { pluck } from 'rxjs/operators';
+
+const mutateRegister = gql`
+  mutation($username: String!, $password: String!, $role: Int!) {
+    register(username: $username, password: $password, role: $role) {
+      id
+      username
+      role
+      token
+
+    }
+  }
+`;
+const mutateLogin = gql`
+  mutation($username: String!, $password: String!) {
+    login(username: $username, password: $password) {
+      id
+      username
+      role
+      token
+    }
+  }
+`;
+export interface UserData {
+  id: string;
+  role: number;
+  token: string;
+  username: string;
+}
```

Comment:
same for interfaces
it's better to export them from either types.ts file or file named according to interface